File: swift/10.swift
import Foundation

func isPrime(_ n: Int) -> Bool {
  if n % 3 == 0 {
    return false
  }

  let lim = Int(sqrt(Double(n)))
  var f = 5
  while f <= lim {
    if n % f == 0 {
      return false
    }

    if n % (f + 2) == 0 {
      return false
    }

    f += 6
  }

  return true
}

var sum = 2 + 3 + 5 + 7
for i in stride(from: 11, to: 2_000_000, by: 2) {
  if isPrime(i) {
    sum += i
  }
}

print(sum)
// 142_913_828_922

File: julia/10.jl
function isPrime(n)
	for i = 2:isqrt(n)
		if n%i === 0
			return false
		end
	end

	return true
end

sum = 0
for i = 2:2_000_000
	if isPrime(i)
		global sum += i
	end
end

println(sum)
# 142913828922
